spotify_dl/spotify.py

```diff
@@ -13,110 +14,200 @@ def fetch_tracks(sp, item_type, url):
+                    if len(item["track"]["album"]["images"]) > 0:
+                        cover = item["track"]["album"]["images"][0]["url"]
                     else:
                         cover = None
 
-                    artists = track_info.get('artists')
-                    main_artist_id = artists[0].get('uri', None) if len(artists) > 0 else None
-                    genres = sp.artist(artist_id=main_artist_id).get('genres', []) if main_artist_id else []
+                    artists = track_info.get("artists")
+                    main_artist_id = (
+                        artists[0].get("uri", None) if len(artists) > 0 else None
+                    )
+                    genres = (
+                        sp.artist(artist_id=main_artist_id).get("genres", [])
+                        if main_artist_id
+                        else []
+                    )
                     if len(genres) > 0:
                         genre = genres[0]
                     else:
                         genre = ""
-                    songs_list.append({"name": track_name, "artist": track_artist, "album": track_album, "year": track_year,
-                                        "num_tracks": album_total, "num": track_num, "playlist_num": offset + 1,
-                                        "cover": cover, "genre": genre, "spotify_id": spotify_id})
+                    songs_list.append(
+                        {
+                            "name": track_name,
+                            "artist": track_artist,
+                            "album": track_album,
+                            "year": track_year,
+                            "num_tracks": album_total,
+                            "num": track_num,
+                            "playlist_num": offset + 1,
+                            "cover": cover,
+                            "genre": genre,
+                            "spotify_id": spotify_id,
+                            "track_url": track_url,
+                        }
+                    )
                     offset += 1
-                    progress.update(task_id=track_info_task, description=f"Fetching track info for \n{track_name}",advance=1)
-                progress.update(task_id=songs_task, description=f"Fetched {offset} of {total_songs} songs from the playlist", advance=100, total=total_songs)
+                    songs_fetched += 1  # added
+                    progress.update(
+                        task_id=track_info_task,
+                        description=f"Fetching track info for \n{track_name}",
+                        advance=1,
+                    )
+                # false info conveyed about numer of songs fetched..is when theres a podcast in list
+                # all songs are feteched but the podcast
+                progress.update(
+                    task_id=songs_task,
+                    description=f"Fetched {songs_fetched} of {total_songs} songs from the playlist",
```

Comment:
added songs fetched variable to keep not of the urls left out in case a podcast url is encountered